cpp/tests/io/text/data_chunk_source_test.cpp

```diff
@@ -342,4 +342,33 @@ TEST_F(DataChunkSourceTest, BgzipCompressedSourceVirtualOffsets)
   test_source(input, *source);
 }
 
+TEST_F(DataChunkSourceTest, BgzipSourceVirtualOffsetsSingleCompressedGZipBlock)
+{
+  auto const filename = temp_env->get_temp_filepath("bgzip_source_offsets_single_compressed_block");
+  std::string const input{"collection unit brings"};
+  std::string head_garbage{"garbage"};
+  // make the head_garbage large enough so the local offset will overrun the compressed block size
+  for (int i = 0; i < 10; i++) {
+    head_garbage += head_garbage;
+  }
+  std::string const tail_garbage{"GARBAGE"};
+  std::size_t begin_compressed_offset{};
+  std::size_t end_compressed_offset{};
+  std::size_t const begin_local_offset{head_garbage.size()};
+  std::size_t const end_local_offset{head_garbage.size() + input.size()};
+  {
+    std::ofstream output_stream{filename};
+    cudf::io::text::detail::bgzip::write_compressed_block(output_stream,
+                                                          head_garbage + input + tail_garbage);
+    cudf::io::text::detail::bgzip::write_uncompressed_block(output_stream, {});
+  }
+
+  auto const source =
+    cudf::io::text::make_source_from_bgzip_file(filename,
+                                                begin_compressed_offset << 16 | begin_local_offset,
```

Comment:
It looks like `begin_compressed_offset` is being default initialized as zero and is never modified (should be `const`?). So is it equivalent to drop that entirely, or am I missing something?
```suggestion
                                                begin_local_offset,
```
Also I had to check the operator precedence between `<<` and `|`, maybe worth using parens for clarity if this is required, like `(a << b) | c`.

Reply:
With the virtual offsets being bit-packed from a 48 bit compressed block offset and a 16 bit local offset, I would prefer keeping the computations here explicit, even if it's just a zero value. I agree with the const-ness of course, that was copypasta from the larger test cases.

Reply:
Explicitly unused computations might deserve a code comment to explain their purpose.

Reply:
WDYT about adding a virtual_offset function that wraps these computations? See latest changes

Reply:
That's a great move. I strongly prefer using that function.